### PR TITLE
Change the IOC name to standard PLC naming

### PR DESCRIPTION
**Pull request by nrwslac**
## Description

- Changed the IOC directory name to standard PLC naming
`ioc-HOMS-XRT-PLC` -> `ioc-xrt-optics`
- the old IOC running on `ioc-xrt-wave8` on port `39007` was deleted and replaced  with the newly named IOC.
- updated `PROJECT_NAME` in iocboot/Makefile
- updated `PREFIX` in iocboot/Makefile
- ran `pre-commit` on all files. (lots of changes, first time on this project)
-   found: LibraryReference Include="LCLS General,newest,SLAC
    changed:LibraryReference Include="LCLS General,2.8.1,SLAC
- upgraded ADS 0.6.0 -> 0.6.1
- upgrade PMPS 3.0.0 -> 3.0.14

## Motivation and Context

Makes searching for the IOC intuitive.

## How Has This Been Tested?

IOC deployed, small test moves made.

## Where Has This Been Documented?

Here

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive comments
- [x] Test suite passes locally
- [x] Libraries are set to fixed versions and not ``Always Newest``
- [x] Code committed with ``pre-commit`` (alternatively ``pre-commit run --all-files``)

### CODE REVIEW

File: iocBoot/ioc-xrt-optics/st.cmd [hunk 1]
```diff
@@ -56,7 +93,7 @@ cd "$(ADS_IOC_TOP)/db"
 
 
 epicsEnvSet("MOTOR_PORT",     "PLC_ADS")
-epicsEnvSet("PREFIX",         "PLC:XRT:HOMS:")
+epicsEnvSet("PREFIX",         "PLC:XRT:OPTICS:")
```
Comment (ZLLentz) [inline iocBoot/ioc-xrt-optics/st.cmd:96]: Can you double confirm that this is live already? If so, I'll modify the pmps config files to point to the new prefix.

**Review by ZLLentz — APPROVED** [review]
All the changes look good to me, both the automatic and manual ones.
Just two quick questions from me

File: iocBoot/ioc-xrt-optics/st.cmd (hunk 1 above)
Comment (nrwslac) [reply iocBoot/ioc-xrt-optics/st.cmd:96]: I can confirm that PLC:XRT:OPTICS is live: 
```
nrw@psbuild-rhel7-01:/reg/g/pcds/gateway/config$ caget PLC:XRT:OPTICS:ProjectInfo:PMPS
PLC:XRT:OPTICS:ProjectInfo:PMPS 3.0.0 -> SLAC - LCLS
nrw@psbuild-rhel7-01:/reg/g/pcds/gateway/config$
```

File: iocBoot/ioc-xrt-optics/st.cmd [hunk 2]
```diff
@@ -359,36 +396,89 @@ dbLoadRecords("EthercatMCreadback.template", "PREFIX=$(MOTOR_PREFIX), MOTOR_NAME
...
+dbLoadRecords("TwinCAT_TaskInfo.db", "PORT=$(ASYN_PORT),PREFIX=PLC:XRT:OPTICS,IDX=3")
+dbLoadRecords("TwinCAT_AppInfo.db", "PORT=$(ASYN_PORT), PREFIX=PLC:XRT:OPTICS")
+
+dbLoadRecords("TwinCAT_Project.db", "PREFIX=PLC:XRT:OPTICS,PROJECT=HOMS_XRT.tsproj,HASH=35c601e,VERSION=v1.2.0-123-g35c601e,PYTMC=2.14.1,PLC_HOST=172.21.88.136")
+
+#   lcls-twincat-motion: * -> 1.8.0 (SLAC)
+dbLoadRecords("TwinCAT_Dependency.db", "PREFIX=PLC:XRT:OPTICS,DEPENDENCY=lcls-twincat-motion,VERSION=1.8.0,VENDOR=SLAC")
+#   lcls-twincat-optics: * -> 0.4.1 (SLAC)
+dbLoadRecords("TwinCAT_Dependency.db", "PREFIX=PLC:XRT:OPTICS,DEPENDENCY=lcls-twincat-optics,VERSION=0.4.1,VENDOR=SLAC")
+#   PMPS: * -> 3.0.0 (SLAC - LCLS)
```
[the remark this answers is not in the document]
Comment (nrwslac) [reply iocBoot/ioc-xrt-optics/st.cmd:415]: I also don't know the answer to this. I guess when the LFE SC update took place 3.0.0 was the first SC updated version. There must have been some quick updates after this; quick enough to leave 3.0.0 untagged. Should we upgrade it?

[the remark this answers is not in the document]
Comment (ZLLentz) [reply iocBoot/ioc-xrt-optics/st.cmd:415]: My gut instinct is we should update but maybe Maggie remembers better

File: iocBoot/ioc-xrt-optics/st.cmd (hunk 1 above)
Comment (ZLLentz) [reply iocBoot/ioc-xrt-optics/st.cmd:96]: Ok I've updated the PMPS config but the new PVs are not live. Did you rebuild the IOC yet and/or is there some other places in the code that need an update?

Comment (nrwslac) [reply iocBoot/ioc-xrt-optics/st.cmd:96]: I may be misunderstanding what you mean by live. The PVs are accessible in the IOC console and as shown in an example from psbuild. I'll need further guidance to demonstrate live.

Comment (ZLLentz) [reply iocBoot/ioc-xrt-optics/st.cmd:96]: Here's an example PV that still uses the old prefix:
```
$ caget -S PLC:XRT:OPTICS:ARB:01:AP:Entry:001:Device_RBV
Channel connect timed out: 'PLC:XRT:OPTICS:ARB:01:AP:Entry:001:Device_RBV' not found.
$ caget -S PLC:XRT:HOMS:ARB:01:AP:Entry:001:Device_RBV
PLC:XRT:HOMS:ARB:01:AP:Entry:001:Device_RBV XRT HOMS
```
In the iocdata PV dump, I see a mix of old and new prefixes
```
/cds/data/iocData/ioc-xrt-optics/iocInfo/IOC.pvlist
```
This implies that something else also needs to update or that something didn't get built

Comment (ZLLentz) [reply iocBoot/ioc-xrt-optics/st.cmd:96]: Ahh maybe the pragmas here are to blame: https://github.com/pcdshub/lcls-plc-xrt-optics/blob/da466536e76b22197849d5d4202a9fb7cd2176d7/HOMS_XRT/HOMS_XRT_PLC/GVLs/GVL_PMPS.TcGVL

Comment (nrwslac) [reply iocBoot/ioc-xrt-optics/st.cmd:96]: yes, that looks like an issue. Good catch. I didn't think to check the pragmas. Let me do that now.

File: HOMS_XRT/HOMS_XRT_PLC/GVLs/GVL.TcGVL [hunk 3]
```diff
@@ -10,11 +10,11 @@
 
     //MPS Outputs
```
Comment (nrwslac) [inline HOMS_XRT/HOMS_XRT_PLC/GVLs/GVL.TcGVL:11]: Should check if this causes problems downstream for MPS.

Comment (nrwslac) [reply HOMS_XRT/HOMS_XRT_PLC/GVLs/GVL.TcGVL:11]: I don't see these PVs in the MPS GUI, so will assume its okay.

File: iocBoot/ioc-xrt-optics/st.cmd (hunk 1 above)
Comment (nrwslac) [reply iocBoot/ioc-xrt-optics/st.cmd:96]:
```
ioc-xrt-optics> dbpr PLC:XRT:HOMS:ARB:01:AP:Entry:001:Device_RBV
PV 'PLC:XRT:HOMS:ARB:01:AP:Entry:001:Device_RBV' not found
ioc-xrt-optics> dbpr PLC:XRT:OPTICS:ARB:01:AP:Entry:001:Device_RBV
ASG :               BUSY: 0             DESC: Assertion Pool                    
DISA: 0             DISP: 0             DISV: 1             
NAME: PLC:XRT:OPTICS:ARB:01:AP:Entry:001:Device_RBV         NORD: 80            
SEVR: NO_ALARM      STAT: NO_ALARM      TPRO: 0             VAL : PTR (nil)     
ioc-xrt-optics> 
```

Comment (nrwslac) [reply iocBoot/ioc-xrt-optics/st.cmd:96]: I think we're good to go now. re-deployed IOC.

File: iocBoot/ioc-xrt-optics/st.cmd (hunk 2 above)
[the remark this answers is not in the document]
Comment (klauer) [reply iocBoot/ioc-xrt-optics/st.cmd:415]: @ghalym is lcls-twincat-pmps 3.0.0 OK to use?

**Comment by klauer** [comment]
Looks like there's a bit more work to do here - Maggie indicated the PMPS library should be updated (so everything needs rebuilding, sorry...!)

**Comment by nrwslac** [comment]
> Looks like there's a bit more work to do here - Maggie indicated the PMPS library should be updated (so everything needs rebuilding, sorry...!)

will set to 3.0.14

**Comment by nrwslac** [comment]
The IOC is rebuilt, I think this is good to go unless anyone has concern about those MPS PV names.

**Review by klauer — APPROVED** [review]

File: iocBoot/ioc-xrt-optics/st.cmd (hunk 1 above)
Comment (ZLLentz) [reply iocBoot/ioc-xrt-optics/st.cmd:96]: Confirming that I see the new PVs and that I've updated the live pmps-ui configuration to use them!

**Comment by ZLLentz** [comment]
What exactly are those MPS PVs used for if they aren't even in the MPS GUI?
Either way I don't see much of an issue here, the PR looks good :+1:

**Pull request merged**